network/src/commonMain/kotlin/com/wire/kalium/network/session/UserSessionManager.kt

```diff
@@ -0,0 +1,45 @@
+package com.wire.kalium.network.session
+
+import com.wire.kalium.network.api.SessionDTO
+import com.wire.kalium.network.api.auth.AccessTokenApiImpl
+import com.wire.kalium.network.api.model.AccessTokenDTO
+import com.wire.kalium.network.api.model.RefreshTokenDTO
+import com.wire.kalium.network.tools.BackendConfig
+import com.wire.kalium.network.utils.NetworkResponse
+import io.ktor.client.HttpClientConfig
+import io.ktor.client.plugins.auth.Auth
+import io.ktor.client.plugins.auth.providers.BearerTokens
+import io.ktor.client.plugins.auth.providers.bearer
+
+interface UserSessionManager {
+    fun userConfig(): Pair<SessionDTO, BackendConfig>
+    fun updateSession(accessToken: AccessTokenDTO, refreshTokenDTO: RefreshTokenDTO?): SessionDTO
+    fun onSessionExpiry()
+}
```

Comment:
What do you think about dropping the `User` prefix? I think mostly leads to confusing when talking about users in the context of sessions.

```suggestion
interface SessionManager {
    fun session(): Pair<SessionDTO, BackendConfig>
    fun updateSession(accessToken: AccessTokenDTO, refreshTokenDTO: RefreshTokenDTO?): SessionDTO
    fun onSessionExpired()
}
```